### PR TITLE
Change regex to exclude Sass partials in sub-directories

### PR DESCRIPTION
**Pull request by mdotasia**
This pull request refers to #35. I noticed that you exclude Sass partials in "stylesheets", but not in sub-directories. Regex is explained in code comments and rejects partials in sub-directories.

I was not sure how to test this. Let me know if you have an idea what exactly to test in a Clash test.

### CODE REVIEW

**Comment by imathis** [comment]
Excellent! I'll work up a test for it. Thanks for your help!

File: lib/octopress-ink/plugin.rb [hunk 1]
```diff
@@ -400,8 +400,21 @@ def sass
...
+      #   dir/_partial.scss is rejected
+      #   dir/dir/_partial.scss is rejected
+      #   snake_case.scss is not rejected
+      #   dir/snake_case.scss is not rejected
+      #   dir/dir/snake_case.scss is not rejected
+      #
+      # Returns Sass assets not including partials.
       def sass_without_partials
-        sass.reject{|f| f.file =~ /^_/ }
+        sass.reject { |f| f.file =~ /^((\w*\/)*_)/ }
```
Comment (parkr) [inline lib/octopress-ink/plugin.rb]: This looks like it could potentially be very slow. Why not `File.basename(f.file).start_with?('_')`?

Comment (imathis) [inline lib/octopress-ink/plugin.rb]: Nicely done @parkr! Thanks for chipping in.

Comment (parkr) [inline lib/octopress-ink/plugin.rb]: I'm pretty into Ruby, you know.

Comment (imathis) [inline lib/octopress-ink/plugin.rb]: It may be the scotch, but you've got me giggling.

**Comment by mdotasia** [comment]
@parkr @imathis re testing of Jekyll websites, can you point me to projects using tests other than than Clash to test Jekyll websites? I am wondering if people use Rsense + Capybara? While Clash tests are easy to write, I find them hard to understand because because often you need to look at actual directory diffs to understand what is being tested.

**Comment by imathis** [comment]
I've added tests and merged this in. I hadn't assumed subdirectories in in assets before, and while creating a test case for this, I discovered some other issues.

1. If a user copied assets using the `octopess ink copy` command, subdirectories weren't retained.
2. Sass partials in subdirectories were being added to the _site/stylesheets/ and css tags if the user configured stylesheets not to be combined.

So thanks a bunch! Here's a look at [the stuff that I added](https://github.com/octopress/ink/commit/e2d8f86bcebfa5901fa5ac0476af3fce81db5648) after your commits.

**Comment by mdotasia** [comment]
Thanks, for fixing this!

**Pull request closed without merging**